### PR TITLE
Adding proxy support for Webhooks

### PR DESCRIPTION
**Pull request by rene1039**
Use Case:
If a Sorry-Cypress instance is behind a proxy (and needs to be run behind a proxy), theres no way to send webhooks without using proxy. This solution is just an example for Slack Webhook, which worked in our environment. If interested, I can include the solution to other available webhooks.

### CODE REVIEW

File: packages/common/src/hook/types.ts [hunk 1]
```diff
@@ -27,6 +27,9 @@ export enum ResultFilter {
 type BaseHook = {
   hookId: string;
   url: string;
+  proxyUrl;
+  proxyUsername;
+  proxyPassword;
```
Comment (agoldis) [inline packages/common/src/hook/types.ts:32]: It doesn't have any typing, you'd better remove alltogether

File: packages/director/src/lib/hooks/reporters/slack.ts [hunk 2]
```diff
@@ -93,6 +94,20 @@ export async function reportToSlack(
         length: 100,
       }
     )}`;
+  
+  hook.proxyUrl = process.env.PROXY_URL
+  hook.proxyUsername = process.env.PROXY_USERNAME
+  hook.proxyPassword = process.env.PROXY_PASSWORD
+
+  if (hook.proxyUrl && hook.proxyUsername && hook.proxyPassword) {
```
Comment (agoldis) [inline packages/director/src/lib/hooks/reporters/slack.ts:102]: Please use `process.env` directly - no need to assign to `hook` properties

**Comment by rene1039** [comment]
@agoldis Hi Andrew, I adjusted your inline comments as well as creating a MR for Sorry-Cypress Docu: https://github.com/sorry-cypress/gitbook/pull/11

How shall we extract the axios client? Do you have any solution or preferred way for this?

**Comment by agoldis** [comment]
> How shall we extract the axios client?  Do you have any solution or preferred way for this?

like this https://axios-http.com/docs/instance

**Comment by rene1039** [comment]
@agoldis I put the axios.create to the controller, so it will be checked when a new hook arrives.

**Pull request closed without merging**